include/cetl/variable_length_array.hpp

```diff
@@ -804,6 +752,105 @@ class VariableLengthArrayBase
         return capacity_ > capacity_before;
     }
 
+    // +----------------------------------------------------------------------+
+    // | CONSTRUCTORS
+    // +----------------------------------------------------------------------+
+
+    constexpr VariableLengthArrayBase(
+        const allocator_type& alloc,
+        value_type*           data,
+        size_type             initial_capacity,
+        size_type             size,
+        size_type             max_size_max) noexcept(std::is_nothrow_copy_constructible<allocator_type>::value)
+        : alloc_(alloc)
+        , data_(data)
+        , capacity_(initial_capacity)
+        , size_(size)
+        , max_size_max_(max_size_max)
+    {
+    }
+
+    constexpr VariableLengthArrayBase(const VariableLengthArrayBase& rhs, const allocator_type& rhs_alloc) noexcept(
+        std::is_nothrow_copy_constructible<allocator_type>::value)
+        : alloc_(std::allocator_traits<allocator_type>::select_on_container_copy_construction(rhs_alloc))
+        , data_(nullptr)
```

Comment:
Is the actual copying done by the subclass then?

Reply:
yep.